x-pack/test/functional/services/ml/trained_models_table.ts

```diff
@@ -226,6 +239,256 @@ export function TrainedModelsTableProvider(
       await trainedModelsActions.testModelOutput(modelType, inputParams, expectedResult);
     }
 
+    public async openTrainedModelsInferenceFlyout(modelId: string) {
+      await mlCommonUI.invokeTableRowAction(
+        this.rowSelector(modelId),
+        'mlModelsTableRowDeployAction',
+        false
+      );
+      await this.assertDeployModelFlyoutExists();
+    }
+
+    public async setTrainedModelsInferenceFlyoutCustomValue(target: string, customValue: string) {
```

Comment:
I think this method does not belong to this particular service, because it's intended for the trained models table and its actions. could you please create a dedicated service (or page object) for the flyout?